arangod/Agency/State.cpp

```diff
@@ -854,12 +855,35 @@ bool State::loadPersisted() {
   loadOrPersistConfiguration();
 
   if (checkCollection("log") && checkCollection("compact")) {
-    bool lc = loadCompacted();
-    bool lr = loadRemaining();
-    return (lc && lr);
+    index_t lastCompactionIndex = loadCompacted();
```

Comment:
This sets `_readDb` and `_spearhead`, so if we fail here, we probably want to roll this back, too.
Furthermore, `_commitIndex` is set to something non-zero and later code might assume it is zero, if we fall back to empty.

Reply:
No that's fine. We don't roll back that either, when we have an old leader come back, rebuildDBs clears all.